### PR TITLE
Package the include/cooperative_groups/details/ headers

### PR DESCRIPTION
**Pull request by robertmaynard**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Compilation tests of the `cuda-cudart-dev` package exposed this oversight.

### CODE REVIEW

**Comment by robertmaynard** [comment]
> Thanks Robert! Can you add a test for this as well please?
> 
> I can't do a proper suggestion, but something like this should be sufficient I would think: `- test -f $PREFIX/targets/{{ target_name }}/include/cooperative_groups/memcpy_async.h # [linux]` `- test -f $PREFIX/targets/{{ target_name }}/include/cooperative_groups/details/async.h # [linux]`

Done!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -107,13 +107,14 @@ outputs:
...
-      - targets/{{ target_name }}/include/*                     # [linux]
-      - targets/{{ target_name }}/include/cooperative_groups/*  # [linux]
-      - targets/{{ target_name }}/lib/*.so                      # [linux]
-      - Library\include                                         # [win]
-      - Library\lib\cuda.lib                                    # [win]
-      - Library\lib\cudadevrt.lib                               # [win]
-      - Library\lib\cudart.lib                                  # [win]
+      - targets/{{ target_name }}/include/*                             # [linux]
+      - targets/{{ target_name }}/include/cooperative_groups/*          # [linux]
+      - targets/{{ target_name }}/include/cooperative_groups/details/*  # [linux]
```
Comment (jakirkham) [inline recipe/meta.yaml:112]: From local testing on a similar case, this appeared to work. Though let's double check

```suggestion
      - targets/{{ target_name }}/include/**                             # [linux]
```

Comment (jakirkham) [reply recipe/meta.yaml:112]: After further offline discussion/exploration, it seems just dropping the `*` fixes this issue. So would suggest doing that here instead

```suggestion
      - targets/{{ target_name }}/include                               # [linux]
```

**Comment by adibbley** [comment]
@conda-forge-admin, please rerender

**Pull request merged**